strawberryfields/backends/states.py

```diff
@@ -359,6 +362,56 @@ def quad_expectation(self, mode, phi=0, **kwargs):
         """
         raise NotImplementedError
 
+    @abc.abstractmethod
+    def poly_quad_expectation(self, A, d=None, k=0, phi=0, **kwargs):
+        r"""The multi-mode expectation values and variance of arbitrary 2nd order polynomials
+        of quadrature operators.
+
+        An arbitrary 2nd order polynomial of quadrature operators over $N$ modes can always
+        be written in the following form:
+
+        .. math:: P(\mathbf{r}) = \frac{1}{2}\mathbf{r}^T A\mathbf{r} + \mathbf{r}^T \mathbf{d} + k I
+
+        where:
+
+        * :math:`A\in\mathbb{R}^{2N\times 2N}` is a symmetric matrix
+          representing the quadratic coefficients,
+        * :math:`\mathbf{d}\in\mathbb{R}^{2N}` is a real vector representing
```

Comment:
In the PR overview, it says that d is a 2N-dimensional complex vector, not real. Just to confirm, it should be real?

Reply:
Yes I guess it should be, otherwise the observable would be non-Hermitian